applications/machine_tending/skills.py

```
@@ -0,0 +1,140 @@
+    pin: int = Field(default=0, description="Pin number for gripper control if not using robotiq gripper")
+
+class machine_tending_implementation(BaseTool):
+    """Tool to perform machine tending operations on cnc."""
+
+    name: ClassVar[str] = "machine_tending_tool"
+    description: ClassVar[str] = "Tool to perform machine tending operation/application."
+    args_schema: ClassVar[Type[BaseModel]] = machine_tending_definition
+    return_direct: ClassVar[bool] = False
+    verbose: ClassVar[bool] = False
+
+    def _run(self, ref_pose: str = "home_pose", object_name: Optional[str] = None) -> bool:
```

Ruff (0.12.2)

`Optional` may be undefined, or defined from star imports

(F405)